File: 347-top-k-frequent-elements/347-top-k-frequent-elements.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        m = Counter(nums)
        minHeap = []
        for num, v in m.items():
            if len(minHeap) < k:
                heapq.heappush(minHeap, (v,num))
            elif v > minHeap[0][0]:
                heapq.heappop(minHeap)
                heapq.heappush(minHeap, (v,  num))
        res = []
        while(minHeap):
            res.append(minHeap[0][1])
            heapq.heappop(minHeap)
        return res
        

File: 347-top-k-frequent-elements/347-top-k-frequent-elements.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> res;
        unordered_map<int, int> m;
        for(int i : nums){
            if(m.find(i) != m.end()){
                m[i] += 1;
            }
            else{
                m[i] = 1;
            }
        }
        priority_queue<pair<int,int>, vector<pair<int, int>>, greater<pair<int, int>>> q;
        for(auto i : m){
            if(q.size() < k){
                q.push(make_pair(i.second, i.first));
            }
            else{
                if(i.second > q.top().first){
                    q.pop();
                    q.push(make_pair(i.second, i.first));
                }
            }
        }
        while(!q.empty()){
            res.push_back(q.top().second);
            q.pop();
        }
        return res;
    }
};